File: src/main/java/com/vaadin/demo/component/slider/SliderStep.java
package com.vaadin.demo.component.slider;

import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.slider.Slider;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("slider-step")
public class SliderStep extends Div {

    public SliderStep() {
        // tag::snippet[]
        Slider slider = new Slider("Volume", 0, 10, 5);
        slider.setStep(0.5);
        add(slider);
        // end::snippet[]
    }

    public static class Exporter extends DemoExporter<SliderStep> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/slider/slider-step.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/slider';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('slider-step')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-slider label="Volume" min="0" max="10" value="5" step="0.5"></vaadin-slider>
      <!-- end::snippet[] -->
    `;
  }
}
